content_resolver/analyzer.py

```diff
@@ -15,6 +17,315 @@ def pkg_placeholder_name_to_nevr(placeholder_name):
     return placeholder_id
 
 
+###############################################################################
+### Standalone functions for ProcessPoolExecutor ############################
+###############################################################################
+
+def _get_build_deps_from_a_root_log_standalone(root_log):
+    """
+    Standalone version of _get_build_deps_from_a_root_log for ProcessPoolExecutor.
+    Must not reference class methods or instance variables.
+    """
+    required_pkgs = []
+
+    # The individual states are nicely described inside the for loop.
+    # They're processed in order
+    state = 0
+    
+    for file_line in root_log.splitlines():
+
+        # 0/
+        # parts of the log I don't really care about
+        if state == 0:
+
+            # The next installation is the build deps!
+            # So I start caring. Next state!
+            if "'builddep', '--installroot'" in file_line:
+                state += 1
+        
+
+        # 1/
+        # getting the "already installed" packages to the list
+        elif state == 1:
+
+            # "Package already installed" indicates it's directly required,
+            # so save it.
+            # DNF5 does this after "Repositories loaded" and quotes the NVR;
+            # DNF4 does this before "Dependencies resolved" without the quotes.
+            if "is already installed." in file_line:
+                # Format: "Package bash-5.1.0-1.fc34.x86_64 is already installed."
+                words = file_line.split()
+                if len(words) >= 4 and words[0] == "Package" and words[-2] == "already":
+                    pkg_nevr = words[1].strip('"')
+                    pkg_name = pkg_nevr.rsplit("-", 2)[0]
+                    required_pkgs.append(pkg_name)
+
+            # That's all! Next state! (DNF4)
+            elif "Dependencies resolved." in file_line:
+                state += 1
+
+            # That's all! Next state! (DNF5)
+            elif "Repositories loaded." in file_line:
+                state += 1
+
+
+        # 2/
+        # going through the log right before the first package name
+        elif state == 2:
+
+            # "Package already installed" indicates it's directly required,
+            # so save it.
+            # DNF4 does this before "Dependencies resolved" without the quotes;
+            # DNF5 does this after "Repositories loaded" and quotes the NVR, but
+            # sometimes prints this in the middle of a dependency line.
+            if "is already installed." in file_line:
+                words = file_line.split()
+                if "already" in words and len(words) >= 4:
+                    pkg_index = words.index("already") - 2
+                    if pkg_index >= 0:
+                        pkg_nevr = words[pkg_index].strip('"')
+                        pkg_name = pkg_nevr.rsplit("-", 2)[0]
```

Comment:
```suggestion
                pkg_index = file_line.split().index("already") - 2
                pkg_name = file_line.split()[pkg_index].strip('"').rsplit("-",2)[0]
```

Reply:
I was debating on this and I think you are right. What you have isn't quite the full change, but I'll revert it back to the original.

Reply:
Done